### ISSUE TITLE
Could not load file or assembly 'System.IO.FileSystem.AccessControl' after upgrading to .NET Core 2.0

### ISSUE THREAD

**Issue by mkosieradzki** [Azure/service-fabric-issues#639]
First of all I would like to thank you all for fixing *all* the blockers related to .NET Core 2.0.

I have hit another issue (non-blocking):
For some my services I am getting:
```
Unhealthy event: SourceId='System.RA', Property='ReplicaOpenStatus', HealthState='Warning', ConsiderWarningAsError=false.
Replica had multiple failures during open on _Node_0. API call: IStatefulServiceReplica.Open(); Error = System.IO.FileNotFoundException (-2147024894)
Could not load file or assembly 'System.IO.FileSystem.AccessControl, Version=4.0.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
   at System.Signature.GetSignature(Void* pCorSig, Int32 cCorSig, RuntimeFieldHandleInternal fieldHandle, IRuntimeMethodInfo methodHandle, RuntimeType declaringType)
   at System.Reflection.RuntimeMethodInfo.FetchNonReturnParameters()
   at System.Reflection.RuntimeMethodInfo.GetParameters()
   at System.Diagnostics.StackTrace.ToString(TraceFormat traceFormat)
   at System.Exception.GetStackTrace(Boolean needFileInfo)
   at Microsoft.ServiceFabric.Replicator.KtlLogManager.d__19.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceFabric.Replicator.LogManager.d__81.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceFabric.Replicator.LoggingReplicator.d__141.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceFabric.Replicator.LoggingReplicator.d__140.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceFabric.Replicator.DynamicStateManager.d__110.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceFabric.Replicator.TransactionalReplicator.d__79.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceFabric.Replicator.StatefulServiceReplica.d__21.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceFabric.Data.ReliableStateManagerImpl.d__45.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceFabric.Services.Runtime.StatefulServiceReplicaAdapter.d__17.MoveNext()
For more information see: http://aka.ms/sfhealth 
```

Fortunately this one is *trivial* to workaround by just adding `System.IO.FileSystem.AccessControl` 4.4.0 nuget package to this specific project.

**Comment by ashishnegi** [comment]
@mkosieradzki Can you share sample code project which can reproduce the problem ?

**Comment by mkosieradzki** [comment]
@ashishnegi Here it will be difficult, but I will try (can't promise anything), but in other problem I think I can create repro easily.

You must understand that I am not writing project from scratch, but I am porting a huge system to .NET Core 2.0 this is why isolating issues is sometimes very difficult.

**Comment by ashishnegi** [comment]
@mkosieradzki We understand that. But without minimum example, it is difficult to find root cause.
Also, share your SF runtime version, SDK version, Env like Linux or windows machine.
If working example is not possible, please share code snippet where exception is coming and your csproj file.

**Comment by mkosieradzki** [comment]
@ashishnegi 

> please share code snippet where exception is coming and your csproj file

This was helpful ;). I have noticed that I have accidentally not switched the project this is happening from net47 to netcoreapp2.0. So this issue seems to affect projects .NET 4.7 projects running SDK 3.0.232-preview.

I think I will be able to isolate the issue now.

**Comment by mkosieradzki** [comment]
@ashishnegi 

OK this is repro (runtime 6.0.232, DevCluster, Windows, SDK 3.0.232-preview)
1. Create Stateful Service in .NET Core 2.0
2. Edit csproj;
- Change `TargetFramework` from netcoreapp2.0 to net47
- Change `Microsoft.AspNetCore.All` to `Microsoft.AspNetCore` with appriopriate version (e.g. 2.0.1)
3. Run service on dev cluster - you will get the aforementioned exception

4. Add `System.IO.FileSystem.AccessControl` 4.4.0, run again and the issue disappears.

**Comment by ashishnegi** [comment]
@mkosieradzki We understand the problem now. We are working actively on this issue.
Can you tell why you are using sdk ` 3.0.232-preview` packages with net47 targeted projects ?

Please use stable `2.8.232` packages with net47 projects for now to unblock yourself. They have all functionalities for net47 projects. 

`3.0.232-preview` are for coreclr targeted projects.

**Comment by mkosieradzki** [comment]
I would like to use a single codebase for net47 and netcoreapp2.0. So especially I need the same SDK for my utility libraries (netstandard2.0).

Currently my netcoreapp2.0 adoption is blocked by other issues.

**Comment by HakanL** [comment]
I'm experiencing this as well, and the workaround is working, but the odd part is that so far we've only run into this issue on two developer machines. We think we have compared all versions, everything is up-to-date, Visual Studio 15.5.6, Service Fabric 3.0.346, etc. What could cause this issue to not happen in all environments? All of these machines have had several versions in the past (they are not clean Windows installs, FWIW).

**Comment by ashishnegi** [comment]
@HakanL and @mkosieradzki 
This is fixed as part of SF version 6.1.472 
Please report if you have any problem.

**Comment by masnider** [comment]
Hi folks. We haven't heard back from you, and we've shipped the fix for this, so we're going to be closing down the issue. If you're still running into problems feel free to reopen this issue or create a new one!
